00-programming-fundamentals/intro-to-oauth.md

```diff
@@ -0,0 +1,46 @@
+
+# Introduction to OAuth
+
+## Learning Goal
+You should be able to:
+*  Explain the OAuth workflow and how an OAuth service can provide authentication and information about a user.
+
+## Introduction
+
+OAuth is an authorization framework, meaning it allows a user to grant access to their user account in one service (the OAuth provider) to another (the client).  At Ada we will primarily use OAuth to use another service (Github) to provide secure authentication to our sites.  
+
+## OAuth Roles
+
+In an OAuth application the following entities play roles in the process.
+
+* The **User** is the person using the browser to access the site.
+* The **Client** is an application which wants access to the user's account.  
+* The **OAuth Provider** is the service providing OAuth access.  It maintains the following services.  
+  * The **Resource Server** is a server managed by the provider which hosts the user accounts.  
+  * The **Authorization Server** verifies the user's identity and provides access tokens to the client.
+
+## Authorization Process
+
+Before the client site can use OAuth from the provider, the client must be registered with the provider.  When the client registers it provides a callback URL, a website (for a web app) and an application name.  In return it receives a **Client ID** and **Client Secret**.  
+
+The **Client ID** is a publicly available string which identifies the service with the API and the **Client Secret** is kept hidden, like a password and used to authenticate with the service.
+
+![OAuth Process](./images/auth_code_flow.png )
```

Comment:
I'm not sure why but right now this is showing as a link and not an image. The image was also not added to the github repo.